arch/arm/core/cortex_m/tz/CMakeLists.txt

```diff
@@ -12,12 +12,24 @@ if(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
 # when building a Non-Secure image which shall have access to Secure
 # Entry functions.
   zephyr_ld_options(
-    ${LINKERFLAGPREFIX},--out-implib=${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
+    ${LINKERFLAGPREFIX},--out-implib=${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
   )
 
   zephyr_ld_options(
     ${LINKERFLAGPREFIX},--cmse-implib
   )
+
+  if(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
```

Comment:
This if-statement always evaluates to true, check out the if-statement above.

Reply:
Oh, of course :). Removed